File: EmployeeManagementAPI/Models/Employee.cs
namespace EmployeeManagementAPI.Models
{
    public class Employee
    {
        public int EmployeeId { get; set; }

        public string FullName { get; set; } = string.Empty;

        public string Email { get; set; } = string.Empty;

        public string Phone { get; set; } = string.Empty;

        public string Gender { get; set; } = string.Empty;

        public int DepartmentId { get; set; }

        public int DesignationId { get; set; }

        public decimal Salary { get; set; }

        public string EmployeeType { get; set; } = string.Empty;
    }
}

File: src/app/models/employee.ts
export interface Employee {
  employeeId: number;
  fullName: string;
  email: string;
  phone: string;
  gender: string;
  departmentId: number;
  designationId: number;
  salary: number;
  employeeType: string;

  departmentName?: string;
  designationName?: string;
}